insights/specs/datasources/lpstat.py

```diff
@@ -43,3 +45,45 @@ def lpstat_protocol_printers_info(broker):
     except Exception as e:
         raise SkipComponent("Unexpected exception:{e}".format(e=str(e)))
     raise SkipComponent
+
+
+@datasource(LocalSpecs.lpstat_o, HostContext)
+def lpstat_queued_jobs_number(broker):
+    """
+    This datasource provides the number of all queued job.
+
+    Typical content of ``/usr/bin/lpstat -o`` file is::
+
+        Cups-PDF-1802           root          265443328   Tue 05 Sep 2023 02:21:19 PM CST
+        Cups-PDF-1803           root          265443328   Tue 05 Sep 2023 02:21:21 PM CST
+        Cups-PDF-1804           root          265443328   Tue 05 Sep 2023 02:21:22 PM CST
+
+    Sample data returned::
+
+        3
+
+    Returns:
+        DatasourceProvider: Returns the collected content containing the number of all queued jobs.
+
+    Raises:
+        SkipComponent: When there is not any content.
+    """
+
+    content = broker[LocalSpecs.lpstat_o].content
+    if content:
+        cnt = 0
+        bad_lines = ["no such file or directory",
+                     "not a directory",
+                     "command not found",
+                     "no module named",
+                     "no files found for",
+                     "missing dependencies:",
+                     "Bad file descriptor"]
+        for line in content:
+            if all(bad_keywords not in line for bad_keywords in bad_lines):
+                cnt += 1
+        if cnt:
+            return DatasourceProvider(content="\n".join([str(cnt)]), relative_path='insights_commands/lpstat_-o')
+        raise SkipComponent
+
+    return DatasourceProvider(content="\n".join(['0']), relative_path='insights_commands/lpstat_-o')
```

Comment:
Better rename the collected file correspondingly, since it's just a number but not the actual `lpstat -o`. How about `lpstat_-o_line_count` ?

Reply:
updated